### DISCUSSION TITLE
ROCm 5.6 Upgrade

### DISCUSSION THREAD

**Discussion by Milor123** [vladmandic/sdnext#1616] [Q&A]
Hi guys, I've seen that in the past days AMD do the new release for 5.6 version of ROCm
![image](https://github.com/vladmandic/automatic/assets/14153649/32232190-dc78-4502-8352-7b23b673ca25)

My question is, it could do a great benefit for my Rx 6700xt, and 6750xt?

Should I upgrade from ROCm 5.4.2 to ROCm 5.6.0? 
I am in ubuntu Studio 22.0.4 LSTC

when I've installed the ROCm, it include **dkms**, i forget use **--no-dkms** 
if the best option is upgrade, could please someone guide me for upgrade it to 5.6 and dont have problem with **dkms**
Thank u very much.

**Comment by vladmandic** [comment]
looping in @lshqqytiger

**Reply by Milor123** [reply]
What do you mean?

I am testing it now (ROCm 5.6 + Pytorch with Nigthly ROMc 5.5). I think that  maybe have a problem while use latente noise in Img2img, but i am not sure.

EDIT: For me all is working nice.

**Reply by Milor123** [reply]
I works very nice for me, 7it/s in 512x512 with Eulear A, text to img
Rx 6700xt

**Reply by marcin-przywoski** [reply]
Did you need to do anything else besides upgrading? I see that in my case it has been detecting that rocm is installed but still not working

**Comment by Milor123** [comment]
@marcin-przywoski 
> Did you need to do anything else besides upgrading? I see that in my case it has been detecting that rocm is installed but still not working

Its is really confuses. in AMD you need 

Install amdgpu package (it package contains the repositories with Propietary, open source and pro drivers, and ROCm Drivers) according to the version of amdgpu package that u select, in some cases you need add or change manually the package in in your linux

I suggest remove all packages and reinstall

`sudo amdgpu-uninstall`

Then 

`sudo amdgpu-install -y --usecase=opencl,graphics,multimedia,workstation,amf,rocm,graphics,hip --vulkan=pro,amdvlk --opencl=rocr --accept-eula --no-32`

If you get error that dont found packages is the STUPID AMD package is because are shit, damn AMD thanks for be a shit, and you need modify the repo, could use 
![image](https://github.com/vladmandic/automatic/assets/14153649/73e90306-48ca-4986-bf77-2bfca602e200)

But in my case with the last drivers works nice, this:
[Radeon™ Software for Linux® version 23.20 for Ubuntu 22.04.3](https://repo.radeon.com/amdgpu-install/23.20/ubuntu/jammy/amdgpu-install_5.7.50700-1_all.deb)

then install torch nightly
`pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/rocm5.6`

or stable version
`pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm5.4.2`

AMD is a painASS, guys never buy AMD or lose the middle of live solving issues, fuck AMD!!!

**Reply by marcin-przywoski** [reply]
Thanks! It made it work for me (just uninstalling and installing again). I'm not sure which version of ROCM I'm using right now but it's still pretty fast, around 9 it/s on this repo and InvokeAI
